Sources/vincenty/vincenty.swift

```diff
@@ -26,101 +40,146 @@ private let pi = Double.pi
         }
-
+        
         sin_alpha = (cos_u_x * cos_u_y * sin(lambda)) / sin_sigma
         cos2_alpha = 1 - sin_alpha * sin_alpha
         cos_2sigma = cos_sigma - (2 * sin_u_x * sin_u_y) / cos2_alpha
-
+        
         // check for nan
         if cos_2sigma.isNaN {
             cos_2sigma = 0.0
         }
-
+        
         c = F / 16.0 * cos2_alpha * (4 + F * (4 - 3 * cos2_alpha))
         tmp = lambda
         lambda = (l + (1 - c) * F * sin_alpha
-            * (sigma + c * sin_sigma
-            * (cos_2sigma + c * cos_sigma
-            * (-1 + 2 * cos_2sigma * cos_2sigma
-        )))
+                  * (sigma + c * sin_sigma
+                     * (cos_2sigma + c * cos_sigma
+                        * (-1 + 2 * cos_2sigma * cos_2sigma
+                          )))
         )
-
+        
         if fabs(lambda - tmp) < tol {
             break
         }
-
+        
     }
     let eps: Double = fabs(lambda - tmp)
     if eps >= tol {
-        throw ConvergenceError.notConverged(maxIter: maxIter, tol: tol, eps: eps)
+        throw VincentyError.notConverged(maxIter: maxIter, tol: tol, eps: eps)
     }
-
+    
     let uu: Double = cos2_alpha * C
     let a: Double = 1 + uu / 16384 * (4096 + uu * (-768 + uu * (320 - 175 * uu)))
     let b: Double = uu / 1024 * (256 + uu * (-128 + uu * (74 - 47 * uu)))
     let delta_sigma: Double = (b * sin_sigma
-        * (cos_2sigma + 1.0 / 4.0 * b
-        * (cos_sigma * (-1 + 2 * cos_2sigma * cos_2sigma)
-        - 1.0 / 6.0 * b * cos_2sigma
-        * (-3 + 4 * sin_sigma * sin_sigma)
-        * (-3 + 4 * cos_2sigma * cos_2sigma))))
+                               * (cos_2sigma + 1.0 / 4.0 * b
+                                  * (cos_sigma * (-1 + 2 * cos_2sigma * cos_2sigma)
+                                     - 1.0 / 6.0 * b * cos_2sigma
+                                     * (-3 + 4 * sin_sigma * sin_sigma)
+                                     * (-3 + 4 * cos_2sigma * cos_2sigma))))
+    
+    let distance = B * a * (sigma - delta_sigma)
+    
+    // note special handling of exactly antipodal points where sin²σ = 0 (due to discontinuity
+    // atan2(0, 0) = 0 but atan2(ε, 0) = π/2 / 90°) - in which case bearing is always meridional,
+    // due north (or due south!)
+    // α = azimuths of the geodesic; α2 the direction P₁ P₂ produced
+    let a1 = abs(sinSq_sigma) < Double.leastNonzeroMagnitude ? 0 : atan2(cos_u_y*sin(lambda),  cos_u_x*sin_u_y-sin_u_x*cos_u_y*cos(lambda))
+    let a2 = abs(sinSq_sigma) < Double.leastNonzeroMagnitude ? Double.pi : atan2(cos_u_x*sin(lambda), -sin_u_x*cos_u_y+cos_u_x*sin_u_y*cos(lambda))
+    
+    let initialTrueTrack = abs(distance) < Double.leastNonzeroMagnitude ? Double.nan : wrap360(a1.toDegrees())
+    let finalTrueTrack = abs(distance) < Double.leastNonzeroMagnitude ? Double.nan : wrap360(a2.toDegrees())
+    
+    return VincentyResults(distance: distance, initialTrueTrack: initialTrueTrack, finalTrueTrack: finalTrueTrack)
```

Comment:
Defining a `struct` as return type is fine. I just would check on the naming of the fields.
I am not an expert in the field, but there seem to be several terms like  `initialTrueTrack`, `initialBearing`, ...
From my perspective the term `azimuths` seem to be the most widespread one. 
So I would suggest to return the result simply as (named) tuple: 
```
    return (distance: distance, azimuths: (initialTrueTrack, finalTrueTrack));
```
This would change the function signature to 
```
-> (distance: Double, azimuths: (Double, Double))
```
which should be quite clear for everyone familiar with vincenty's formula. 
Instead of a named tuple one can also use a struct, but I would group the angles separately from the distance.

Reply:
I used aeronautical terms as that's what I'm used to. Normally you want the output in degrees though.

Reply:
Ok, we're sticking to radians and azimuths.